[bug] "npm create tauri-app" fails on openSUSE Tumbleweed with "Not a directory" error

### Describe the bug

Trying to create a project under openSUSE Tumbleweed with `npm create tauri-app`, it fails with the following error:

```text
error: Not a directory (os error 20)
npm ERR! code 1
npm ERR! path /tmp
npm ERR! command failed
npm ERR! command sh -c -- create-tauri-app

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/x/.npm/_logs/2022-09-28T16_53_00_647Z-debug-0.log
```

Same error with any other tool (e.g. yarn).

### Reproduction

_No response_

### Expected behavior

_No response_

### Platform and versions

```
Environment
  › OS: openSUSE 20220926 X64
  › Node.js: 18.9.1
  › npm: 8.19.1
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.64.0
  › cargo: 1.64.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 
```

### Stack trace

_No response_

### Additional context

I tried running it with `strace` utility and the syscall which fails is:

`openat(AT_FDCWD, "/usr/bin/node18", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOTDIR (Not a directory)`

Node.js installation contains two executables: `/usr/bin/node` (which seems to be a stub) and `/usr/bin/node18` (the real one).
If I rename `/usr/bin/node18` to `/usr/bin/node` then everything works.

This issue is specific to tauri, I tried for example creating other apps like react and it worked.

--- a/.changes/cli-node.md
+++ b/.changes/cli-node.md
@@ -1,0 +1,5 @@
+---
+"create-tauri-app-js": "patch"
+---
+
+Fix crash when nodejs binary has the version in its name, for example `node18`

--- a/packages/cli/node/create-tauri-app.js
+++ b/packages/cli/node/create-tauri-app.js
@@ -16,7 +16,7 @@ let binName;
 
 // Even if started by a package manager, the binary will be NodeJS.
 // Some distribution still use "nodejs" as the binary name.
-if (binStem === "node" || binStem === "nodejs") {
+if (binStem.match(/(nodejs|node)([1-9]*)*$/g)) {
   const managerStem = process.env.npm_execpath
     ? path.parse(process.env.npm_execpath).name.toLowerCase()
     : null;

